cfpb/cfgov-refresh
# Refactoring email signup to remove validate.js

Refactoring email signup to remove validate.js.
## Additions
- Added Email Signup organism code `cfgov/unprocessed/js/organisms/EmailSignup.js` to handle Gov Delivery signups via ajax.
- Added `cfgov/unprocessed/js/routes/on-demand/email-signup.js` to be used for on-demand route handling via `base.html`.
## Removals
- Removed `cfgov/unprocessed/js/modules/form-validation.js`.
- Removed `cfgov/unprocessed/js/modules/jquery/cf_formValidator.js`.
- Removed `cfgov/unprocessed/js/modules/jquery/cf_notifier.js`.
- Removed  `validate.js` from `package.json`
## Changes
- Modified `cfgov/jinja2/v1/_includes/organisms/email-signup.html` to include notification markup.
- Modifeld `cfgov/unprocessed/js/config/error-messages-config.js` to add form submission messages.
- Modified `cfgov/unprocessed/js/modules/util/validators.js` to fix issue with email validation escape character. 
- Modified `cfgov/unprocessed/js/molecules/Notification.js` to add commented out legacy code. The code might be useful in Single Page Applications.
## Testing
- Run `gulp build`
- Visit the blog page.
- Enter `test@blog` in the `Stay Informed` email field. Click `Submit` and verify that you received a error notification.
- Enter `test@blog.com` in the email field. Click `Submit` and verify that you received a success notification.
## Review
- @chosak 
- @anselmbradford 
- @contolini 
## Notes
- The code is dependent up the correct http status code being returned from the server. It doesn't check the response for failures.
## Todos
- Write unit tests.
## Checklist
